### PR TITLE
feat(connect): allow optional params, cosmetic fixes

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

Fixes https://linear.app/nango/issue/NAN-1947/allow-some-fields-to-be-optional

- Allow optional params
- Fix: do not display back on a single integration scenario
- Fix: move help at the bottom of the screen
- Fix: change button wording when error in credentials

### CODE REVIEW

File: packages/connect-ui/src/lib/utils.ts [hunk 1]
```diff
@@ -5,17 +5,29 @@ import { z } from 'zod';
...
+        fieldString = fieldString.url();
+    }
+    if (schema.pattern) {
+        fieldString = fieldString.regex(new RegExp(schema.pattern), { message: `Incorrect ${schema.title}` });
+    }
 
-    return field;
+    if ('optional' in schema && schema.optional === true) {
+        // https://stackoverflow.com/questions/73715295/react-hook-form-with-zod-resolver-optional-field
+        return z.union([fieldString, z.literal('')]);
```
Comment (TBonnin) [inline packages/connect-ui/src/lib/utils.ts:29]: why not `.optional`?

File: packages/connect-ui/src/lib/utils.ts [hunk 2]
```diff
@@ -5,17 +5,29 @@ import { z } from 'zod';
...
+    if (schema.format === 'hostname') {
+        fieldString = fieldString.regex(/^[a-zA-Z0-9-]$/);
+    } else if (schema.format === 'uuid') {
+        fieldString = fieldString.uuid();
+    } else if (schema.format === 'uri') {
+        fieldString = fieldString.url();
+    }
+    if (schema.pattern) {
+        fieldString = fieldString.regex(new RegExp(schema.pattern), { message: `Incorrect ${schema.title}` });
+    }
```
Comment (TBonnin) [inline packages/connect-ui/src/lib/utils.ts:25]: can we have a pattern with a known format? Should it be `else`?

**Review by TBonnin — APPROVED** [review]

File: packages/connect-ui/src/lib/utils.ts (hunk 1 above)
Comment (bodinsamuel) [reply packages/connect-ui/src/lib/utils.ts:29]: it doesn't work, hence the link

File: packages/connect-ui/src/lib/utils.ts (hunk 2 above)
Comment (bodinsamuel) [reply packages/connect-ui/src/lib/utils.ts:25]: no, in the case of URI for example: URI is not enforcing https which we with the pattern

File: packages/connect-ui/src/lib/utils.ts (hunk 1 above)
Comment (TBonnin) [reply packages/connect-ui/src/lib/utils.ts:29]: by bad. I had missed the link 🙈

**Pull request merged**